ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.committer
+
+import com.codahale.metrics.MetricRegistry
+import com.daml.ledger.participant.state.kvutils.Conversions
+import com.daml.ledger.participant.state.kvutils.Conversions.buildTimestamp
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+import com.daml.ledger.participant.state.kvutils.TestHelpers._
+import com.daml.ledger.participant.state.kvutils.committer.TransactionCommitter.DamlTransactionEntrySummary
+import com.daml.ledger.participant.state.v1.Configuration
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.engine.Engine
+import com.daml.metrics.Metrics
+import com.google.protobuf.ByteString
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{Matchers, WordSpec}
+
+class TransactionCommitterSpec extends WordSpec with Matchers with MockitoSugar {
+  private val metrics = new Metrics(new MetricRegistry)
+  private val aDamlTransactionEntry = DamlTransactionEntry.newBuilder
+    .setSubmitterInfo(
+      DamlSubmitterInfo.newBuilder
+        .setCommandId("commandId")
+        .setSubmitter("aSubmitter"))
+    .setSubmissionSeed(ByteString.copyFromUtf8("a" * 32))
+    .build
+  private val aTransactionEntrySummary = DamlTransactionEntrySummary(aDamlTransactionEntry)
+  private val aRecordTime = Timestamp(100)
+
+  "deduplicateCommand" should {
+    "continue if record time is not available" in {
+      val instance =
+        new TransactionCommitter(theDefaultConfig, mock[Engine], metrics, inStaticTimeMode = false)
+      val context = new FakeCommitContext(recordTime = None)
```

Comment:
Added.

Reply:
I think we should test all the valid situations regardless of our knowledge of how the code currently works, so that we are covered against regressions. Good catch RE the other one.

Reply:
That's a good point -- here I was focusing more on testing the risky parts and skipping the trivial parts for now as I didn't see much value spending more time on that.